### PR TITLE
AG - added column called username to leaderboard that maps user id to the correct username

### PR DESCRIPTION
**Pull request by spartan105**
## Overview

1) when a user joins a commons, the UserCommon's repository now saves a username parameter that is the current user's google full name (this can be easily changed to email, firstname, or last name).
2) a username column was added to the leaderboard page that displays all users in common with their username from (1).

### Backend

- UserCommons.java: added username to UserCommons
- CommonsController.java: username gets initialized as current user's name, UserCommons gets built when a player joins
- UserCommonsControllerTests.java: updated all tests
- CommonsControllerTests.java: updated all tests

### Frontend

- LeaderboardTable.js : added the username column to the table
- LeaderboardTable.test.js: updated all tests
- leaderboardFixtures.js: updated fixtures

### CODE REVIEW

**Review by RAILGUN1124 — DISMISSED** [review]
i approve

**Review by nathanmathew02 — CHANGES_REQUESTED** [review]
Merge Conflict

**Review by tkumar07 — DISMISSED** [review]
LGTM

**Review by nathanmathew02 — APPROVED** [review]
LGTM

**Review by ruiqiRichard — APPROVED** [review]
LGTM

**Review by RAILGUN1124 — APPROVED** [review]
lgtm

File: frontend/src/main/components/OurTable.js [hunk 1]
```diff
@@ -4,9 +4,9 @@ import { Table, Button } from "react-bootstrap";
 import Plaintext from "main/components/Utils/Plaintext";
 
 // Stryker disable all
-var tableStyle = {
-  "background": "white"
-};
+// var tableStyle = {
+//   "background": "white"
+// };
```
Comment (pconrad) [inline frontend/src/main/components/OurTable.js:9]: Why are we commenting out these three lines?

This is undoing someone else's work.  Please don't do that.

File: frontend/src/main/components/OurTable.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ export default function OurTable({ columns, data, testid = "testid", ...rest })
   }, useSortBy)
 
   return (
-    <Table style={tableStyle} {...getTableProps()} striped bordered hover >
+    <Table {...getTableProps()} striped bordered hover >
```
Comment (pconrad) [inline frontend/src/main/components/OurTable.js:29]: Why are we undoing someone else's work here?  Please don't!

File: frontend/src/main/pages/PlayPage.js [hunk 3]
```diff
@@ -62,11 +62,7 @@ export default function PlayPage() {
   // Stryker enable all 
 
   const onSuccessBuy = () => {
-	// The value for totalWealth updates late, the current totalWealth is always the previous totalWealth.
-	// Thats why only need to compare the current totalWealth to the cowPrice, because this implies that 
-	// totalWealth - cowPrice aka current totalWealth will not be enough to buy a cow.
-	if(!(userCommons.totalWealth < commons.cowPrice)) toast(`Cow bought!`);
-	else toast(`You can't buy a cow because you don't have enough money`)
+    toast(`Cow bought!`);
```
Comment (pconrad) [inline frontend/src/main/pages/PlayPage.js:65]: Again.. undoing other people's work is  not acceptable.

I think you might need to merge in the latest changes from main.

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java [hunk 4]
```diff
@@ -18,16 +17,17 @@
...
+  private long commonsId;
 
-  @Column(name="commons_id")
-  private long commonsId;  
+  @Column(name = "user_id")
+  private long userId;
 
-  @Column(name="user_id")
-  private long userId;  
+  private String username;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java:28]: This is not how this should be done.   Instead, you should look up the username from the user's table.  This is storing redundant data, which violates the principles of good database design.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java [hunk 5]
```diff
@@ -49,394 +48,394 @@
...
-
-    verify(commonsRepository, times(1)).save(commons);
-
-    String actualResponse = response.getResponse().getContentAsString();
-    assertEquals(expectedResponse, actualResponse);
-  }
+    @MockBean
+    UserCommonsRepository userCommonsRepository;
+
+    @MockBean
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:54]: If this is only whitespace/formatting changes, PLEASE DON'T DO THIS.

It makes your PR super hard to code review, because we have to look at every blessed line to make sure you didn't delete or add something.

Fixing formatting should be done in a separate PR that ONLY focuses on fixing formatting.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 6]
```diff
@@ -44,40 +44,42 @@
 @AutoConfigureDataJpa
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:44]: If this is only whitespace/formatting changes, PLEASE DON'T DO THIS.

It makes your PR super hard to code review, because we have to look at every blessed line to make sure you didn't delete or add something.

Fixing formatting should be done in a separate PR that ONLY focuses on fixing formatting.

**Review by pconrad — CHANGES_REQUESTED** [review]
This PR has many issues that need to be addressed before it is mergeable.

It does work... I tested it on QA.  But it may very well undo several other changes, and therefore break things that were already fixed.

And it includes huge formatting changes that make it very difficult to review.  Those would be considered quite a nuisance in a professional shop.   Formatting updates are welcome, but should be isolated in their own separate PRs.

**Comment by ruiqiRichard** [comment]
issues get fixed in #65
